src/coreclr/tools/Common/TypeSystem/Common/AsyncMethodVariant.cs

```diff
@@ -0,0 +1,88 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Internal.TypeSystem.Ecma;
+
+namespace Internal.TypeSystem
+{
+    /// <summary>
+    /// Either the AsyncMethodImplVariant or AsyncMethodThunkVariant of a method marked .IsAsync.
```

Comment:
```suggestion
    /// MethodDesc that represents  async calling convention entrypoint of a Task-returning method.
```
The method needs to be Task-returning, it does not need to be marked .IsAsync.